scheduler/src/cook/mesos/scheduler.clj

```diff
@@ -634,12 +633,31 @@
 (defn pending-jobs->considerable-jobs
   "Limit the pending jobs to considerable jobs based on usage and quota.
    Further limit the considerable jobs to a maximum of num-considerable jobs."
-  [db pending-jobs user->quota user->usage num-considerable pool-name]
+  [db pending-jobs user->quota user->usage num-considerable pool-name->optimizer-suggested-job-ids-atom pool-name]
   (log/debug "In" pool-name "pool, there are" (count pending-jobs) "pending jobs:" pending-jobs)
-  (->> pending-jobs
-       (filter-based-on-quota user->quota user->usage)
-       (filter (fn [job] (util/job-allowed-to-start? db job)))
-       (take num-considerable)))
+  (let [optimizer-schedule-job-ids-promise (promise)]
+    (swap! pool-name->optimizer-suggested-job-ids-atom
+           (fn [pool-name->optimizer-schedule-job-ids]
+             (->> (pool-name->optimizer-schedule-job-ids pool-name)
+                  (deliver optimizer-schedule-job-ids-promise))
+             ;; TODO determine whether we should be clearing out optimizer data after one use?
+             (assoc pool-name->optimizer-schedule-job-ids pool-name [])))
```

Comment:
We should probably keep the data around for more than one cycle, but not indefinitely.

Since Fenzo isn't guaranteed to see enough resources to match all of the optimizer-recommended jobs, it seems like a terrible waste to pass them to Fenzo just once and then throw them away.

But conversely, the optimizer runs on a different order of magnitude of frequency from the fenzo-matching loop, so we really don't want a bunch of "old news" from the optimizer to stick around until it spits out a new recommended schedule.

We'll need to find a way to keep them around for a while, and then "expire" them or something. Maybe we can put a timestamp in the scheduler output, and ignore it if it's too old?

Reply:
There's also the issue that we need to remove entries from the optimizer schedule once they get matched by Fenzo, otherwise they continue to take valuable slots from the `num-considerable` limit even after they're running.